activitysim/test/random_seed/test_random_seed.py

```diff
@@ -0,0 +1,127 @@
+
+from activitysim.core import inject
+
+
+def update_settings(settings_file, key, value):
+    with open(settings_file, "r") as f:
+        settings = yaml.safe_load(f)
+        f.close()
+
+    settings[key] = value
+
+    with open(settings_file, "w") as f:
+        yaml.safe_dump(settings, f)
+        f.close()
+
+
+def run_test_random_seed():
+
+    steps_to_run = [
+        "initialize_landuse",
+        "initialize_households",
+        "compute_accessibility",
+        "workplace_location",
+        "write_tables",
+    ]
+
+    def example_path(dirname):
+        resource = os.path.join("examples", "prototype_mtc", dirname)
+        return pkg_resources.resource_filename("activitysim", resource)
+
+    def test_path(dirname):
+        return os.path.join(os.path.dirname(__file__), dirname)
+
+    def create_rng_configs(rng_base_seed=None):
+        new_configs_dir = test_path("configs_random_seed_=_{}".format(rng_base_seed))
+        new_settings_file = os.path.join(new_configs_dir, "settings.yaml")
+        os.mkdir(new_configs_dir)
+        for f in os.listdir(example_path("configs")):
+            copy(os.path.join(example_path("configs"), f), new_configs_dir)
+
+        update_settings(new_settings_file, "models", steps_to_run)
+        if rng_base_seed != "":  # Undefined
+            update_settings(new_settings_file, "rng_base_seed", rng_base_seed)
+
+    # (run name, rng_base_seed value)
+    runs = [
+        ("0-a", 0),
+        ("0-b", 0),
+        ("1-a", 1),
+        ("1-b", 1),
+        ("None-a", None),
+        ("None-b", None),
+        ("Undefined", ""),
+    ]
+
+    seeds = list(set([run[1] for run in runs]))
+    for seed in seeds:
+        create_rng_configs(seed)
+
+    outputs = {}
+
+    def check_outputs(df1, df2, should_be_equal=True):
+        """
+        Compares df1 and df2 and raises an AssertionError if they are unequal when `should_be_equal` is True and equal when `should_be_equal` is False
+        """
+        if should_be_equal:
+            pdt.assert_frame_equal(outputs[df1], outputs[df2])
+        else:
+            try:
+                pdt.assert_frame_equal(outputs[df1], outputs[df2])
+            except AssertionError:
+                pass
+            else:
+                raise AssertionError
+
+    file_path = os.path.join(os.path.dirname(__file__), "simulation.py")
+
+    for name, seed in runs:
+
+        run_args = [
+            "-c",
+            test_path("configs_random_seed_=_{}".format(seed)),
+            "-d",
+            example_path("data"),
+            "-o",
+            test_path("output"),
+        ]
+
+        try:
+            os.mkdir(test_path("output"))
+        except FileExistsError:
+            pass
+
+        subprocess.run(["coverage", "run", "-a", file_path] + run_args, check=True)
+
+        # Read in output to memory to compare later
+        outputs[name] = pd.read_csv(
+            os.path.join(test_path("output"), "final_persons.csv")
```

Comment:
Final persons is being checked for the test, but the above settings file is only outputting trips... Is the right settings.yanl file being read?  Not sure how this is working.

Reply:
The test isn't reading this settings file. There's a function `create_rng_configs` that creates a settings file for a specified random seed by copying the settings file stored in the prototype_mtc example and editing it to use the specified random seed (as well as removing all of the model steps not needed for the test). Each run of ActivitySim in the test then calls the appropriate settings file. I can change it to copy this settings file or remove this configs folder to reduce confusion.